src/main/java/com/code_intelligence/jazzer/utils/UnsafeUtils.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2023 Code Intelligence GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.code_intelligence.jazzer.utils;
+
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodHandles.Lookup;
+import java.lang.reflect.Array;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.Arrays;
+import java.util.Optional;
+import org.objectweb.asm.ClassWriter;
+import org.objectweb.asm.Opcodes;
+
+public final class UnsafeUtils {
+  /**
+   * Dynamically creates a concrete class implementing the given abstract class.
+   *
+   * <p>The returned class will not be functional and should only be used to construct instances
+   * via {@link sun.misc.Unsafe#allocateInstance(Class)}.
+   */
+  public static <T> Class<? extends T> defineAnonymousConcreteSubclass(Class<T> abstractClass) {
+    if (!Modifier.isAbstract(abstractClass.getModifiers())) {
+      throw new IllegalArgumentException(abstractClass + " is not abstract");
+    }
+
+    ClassWriter cw = new ClassWriter(0);
+    String superClassName = abstractClass.getName().replace('.', '/');
+    // Only the package of the class name matters, the actual name is generated. defineHiddenClass
+    // requires the package of the new class to match the one of the lookup.
+    String className = UnsafeUtils.class.getPackage().getName().replace('.', '/') + "/Anonymous";
```

Comment:
Don't we have to implement abstract methods to instantiate the subclass? If not, why couldn't the given abstract class be instantiated directly?

Reply:
Turns out that we don't have to implement them - they will throw exceptions when called, but they aren't called. `Unsafe#allocateInstance` does not support abstract classes, which is why we need to provide a concrete subclass.

Reply:
I expected some kind of verification error defining, and even instantiating, a class not implementing all abstract methods. The tests show that it works, still strange, though.